### ISSUE TITLE
colorama does not work in notebook: 'OutStream' object has no attribute 'closed'

### ISSUE THREAD

**Issue by JanSchulz** [ipython/ipython#7764]
Using latest (as in conda) colorama in ipython notebook (master from yesterday) on Win7 does not work anymore:

```
import colorama
colorama.init()
```
results in:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-a141727eea92> in <module>()
      5 
      6 import colorama
----> 7 colorama.init()

C:\portabel\miniconda\envs\monte-carlo\lib\site-packages\colorama\initialise.pyc in init(autoreset, convert, strip, wrap)
     29     else:
     30         sys.stdout = wrapped_stdout = \
---> 31             wrap_stream(orig_stdout, convert, strip, autoreset, wrap)
     32     if sys.stderr is None:
     33         wrapped_stderr = None

C:\portabel\miniconda\envs\monte-carlo\lib\site-packages\colorama\initialise.pyc in wrap_stream(stream, convert, strip, autoreset, wrap)
     59     if wrap:
     60         wrapper = AnsiToWin32(stream,
---> 61             convert=convert, strip=strip, autoreset=autoreset)
     62         if wrapper.should_wrap():
     63             stream = wrapper.stream

C:\portabel\miniconda\envs\monte-carlo\lib\site-packages\colorama\ansitowin32.py in __init__(self, wrapped, convert, strip, autoreset)
     67         if convert is None:
     68             print(wrapped, type(wrapped))
---> 69             convert = on_windows and not wrapped.closed and not on_emulated_windows and is_a_tty(wrapped)
     70         self.convert = convert
     71 

AttributeError: 'OutStream' object has no attribute 'closed'
```

The debugprintout (see the added line 68 in the above trace) prints:
```
(<IPython.kernel.zmq.iostream.OutStream object at 0x0000000003CF4CC0>, <class 'IPython.kernel.zmq.iostream.OutStream'>)
```

The python API seems to say that this property should exist on streams  (https://docs.python.org/2/library/io.html#io.IOBase.closed) so I suspect that this is an IPython issue and I'm filing this issue here. Please tell me if I should go to the colorama bugtracker...

**Comment by JanSchulz** [comment]
Note that the above page also says that `An IOError is raised if the IO object does not use a file descriptor.` but OutStream raises an `UsupportedOperation` exception (https://github.com/ipython/ipython/blob/master/IPython/kernel/zmq/iostream.py#L200, see also https://github.com/stan-dev/pystan/issues/3)

**Comment by JanSchulz** [comment]
I added a `closed` property in #7765, which lets me import colorama. Not sure about the `fileno()` Exception (and maybe other properties/methods of the stream)?

**Comment by Carreau** [comment]
`UsupportedOperation` is [an `IOError` and a `ValueError`](https://docs.python.org/3.1/library/io.html#io.UnsupportedOperation), so it should not need changes I suppose.

**Comment by minrk** [comment]
Closed by #7765
